pennylane/workflow/execution.py

```diff
@@ -745,23 +603,21 @@ def cost_fn(params, x):
     elif cache is False:
         cache = None
 
-    expand_fn = _preprocess_expand_fn(expand_fn, device, max_expansion)
-
     # changing this set of conditions causes a bunch of tests to break.
-    no_interface_boundary_required = interface is None or gradient_fn in {None, "backprop"}
+    no_interface_boundary_required = interface is None or config.gradient_method in {
+        None,
+        "backprop",
+    }

```

Comment:
If we're no longer using `gradient_fn`, should that be deprecated as well? @trbromley

Reply:
using gradient_fn to construct an `ExecutionConfig` if none is provided.  Also if `gradient_fn` is a gradient trasnform.